apache/superset
# chore: Improves the Select component UI/UX - iteration 4

### SUMMARY
Improves the Select component UI/UX - iteration 4.
- Migrates the native filters Select components to the new Select
- Fixes some bugs of the Select component related to searching and selection
- Simplifies the Select API for the async non-paginated version

Touched components:
```
src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/ColumnSelect.tsx
src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigForm/FiltersConfigForm.tsx
src/filters/components/common.ts
src/filters/components/GroupBy/GroupByFilterPlugin.tsx
src/filters/components/Select/SelectFilterPlugin.tsx
src/filters/components/TimeColumn/TimeColumnFilterPlugin.tsx
src/filters/components/TimeGrain/TimeGrainFilterPlugin.tsx
```
@rusackas @villebro 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/124596960-40682000-de39-11eb-8ca5-0cb26af8080b.mp4

### TESTING INSTRUCTIONS
- Enter in a dashboard
- Open the filter bar
- Open the native filter modal
- Check the selects of the native filter modal
- After adding some filters check the selects in the filter bar

### ADDITIONAL INFORMATION
